MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,363 @@
+# Migration guide for version 8
+
+## Required upgrade to latest Stripe API version
+ Up until `stripe-java` version 7, we maintained model classes in a backward-compatible
+ way. Regardless of your Stripe API version (in dashboard settings), you could always use the 
+ latest version of this client library. In this major version 8, and moving forward, the library 
+ only supports model classes of latest Stripe API version. To use this new version safely, it 
+ requires your Stripe API version upgrade to the latest one.
+ 
+ Under the hood, this new version 8 sets `Stripe-Version` request header to `Stripe#API_VERSION` 
+ statically-defined in the library. Making API requests alone is fine; the return JSON response 
+ has schema according to the specified version, and it is compatible with Java model 
+ classes of the library. However, event integration without upgrading your API version, can 
+ result in unhandled failures, both for webhooks and `/v1/events/*` via `Event#retrieve`.
+ 
+ The event data object in `Event#data#object` is JSON rendered according to your Stripe API 
+ version at the time of its creation. Without upgrading your API version, using this 
+ library pinned at a specific version has no implication on the JSON schema of the data 
+ object; setting version in the header to retrieve events will only render the outer wrapper and 
+ metadata as the specified version, but not the inner data object. Similarly, the data object in 
+ webhook events are also rendered at your Stripe API version. Thus, simply deserializing this data 
+ object at old API version to the model of latest schema in this library can cause incomplete 
+ deserialized object or runtime exceptions. Therefore, it is necessary that you upgrade to the 
+ latest API version to ensure events are created with the same API version supported by this library. 
+ 
+## API version upgrade with `stripe-java`
+
+According to the official [recommendations](https://stripe.com/docs/upgrades#how-can-i-upgrade-my-api)
```

Comment:
Suggestion:
```
[Upgrading your Stripe API version](https://stripe.com/docs/upgrades#how-can-i-upgrade-my-api) affects:

1. Requests and responses for normal API calls
2. The event response structure

Reply:
I mean to make this guide as additional explanations for Java users, and make reference to the "official guide" again later in explaining the events.